src/EditorFeatures/Test2/InlineParameterNameHints/CSharpInlineParamNameHintsTests.vb

```diff
@@ -0,0 +1,202 @@
+                <Project Language="C#" CommonReferences="true">
+                    <Document>
+class A
+{
+    int testMethod(int x, double y)
+    {
+        return x;
+    }
+    void Main() 
+    {
+        $$testMethod({|x:-5|}, {|y:2|});
+    }
+}
+                    </Document>
+                </Project>
+            </Workspace>
+
+            Await VerifyParamHints(input)
+        End Function
+
+        <WpfFact, Trait(Traits.Feature, Traits.Features.InlineParameterNameHints)>
+        Public Async Function TestLiteralNestedCastParametersSimpleCase() As Task
+            Dim input =
+            <Workspace>
+                <Project Language="C#" CommonReferences="true">
+                    <Document>
+class A
+{
+    int testMethod(int x, double y)
+    {
+        return x;
+    }
+    void Main() 
+    {
+        $$testMethod({|x:(int)(double)(int)5.5|}, {|y:2|});
+    }
+}
+                    </Document>
+                </Project>
+            </Workspace>
+
+            Await VerifyParamHints(input)
+        End Function
+
+        <WpfFact, Trait(Traits.Feature, Traits.Features.InlineParameterNameHints)>
+        Public Async Function TestObjectCreationParametersSimpleCase() As Task
+            Dim input =
+            <Workspace>
+                <Project Language="C#" CommonReferences="true">
+                    <Document>
+class A
+{
+    int testMethod(int x, object y)
+    {
+        return x;
+    }
+    void Main() 
+    {
+        $$testMethod({|x:(int)5.5|}, {|y:new object()|});
+    }
+}
+                    </Document>
+                </Project>
+            </Workspace>
+
+            Await VerifyParamHints(input)
+        End Function
+
+        <WpfFact, Trait(Traits.Feature, Traits.Features.InlineParameterNameHints)>
+        Public Async Function TestCastingANegativeSimpleCase() As Task
+            Dim input =
+            <Workspace>
+                <Project Language="C#" CommonReferences="true">
+                    <Document>
+class A
+{
+    int testMethod(int x, object y)
+    {
+        return x;
+    }
+    void Main() 
+    {
+        $$testMethod({|x:(int)-5.5|}, {|y:new object()|});
+    }
+}
+                    </Document>
+                </Project>
+            </Workspace>
+
+            Await VerifyParamHints(input)
+        End Function
+
+
+        <WpfFact, Trait(Traits.Feature, Traits.Features.InlineParameterNameHints)>
+        Public Async Function TestNegatingACastSimpleCase() As Task
+            Dim input =
+            <Workspace>
+                <Project Language="C#" CommonReferences="true">
+                    <Document>
+class A
+{
+    int testMethod(int x, object y)
+    {
+        return x;
+    }
+    void Main() 
+    {
+        $$testMethod({|x:-(int)5.5|}, {|y:new object()|});
+    }
+}
+                    </Document>
+                </Project>
+            </Workspace>
+
+            Await VerifyParamHints(input)
+        End Function
+    End Class
```

Comment:
3. A test where a parameter has no name.
4. Invocations of a delegate.
5. Invocations of a function pointer.

Reply:
__arglist
params array